Classes/common/CDTFetchChanges.m

```diff
@@ -61,16 +65,23 @@ - (void)main
     self.mDatastore = self.datastore;
     self.mStartSequenceValue = self.startSequenceValue;
 
-    TDChangesOptions options = {.limit = 500,
-        .contentOptions = 0,                                
+    NSUInteger mResultsLimit = self.resultsLimit;
+    BOOL thereIsALimit = (mResultsLimit > 0);
+
+    TDChangesOptions options = {
+        .limit = (thereIsALimit && (mResultsLimit < CDTFETCHCHANGES_DEFAULT_OPTION_LIMIT)
+                      ? (unsigned)mResultsLimit
+                      : CDTFETCHCHANGES_DEFAULT_OPTION_LIMIT),
```

Comment:
I think the setting of `.limit` would be easier to read if it was was a `if` statement.